client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -213,9 +214,8 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionPool con
     private <T> RevisionedStreamClient<T> createRevisionedStreamClient(Segment segment, Serializer<T> serializer,
                                                                        SynchronizerConfig config) {
         EventSegmentReader in = inFactory.createEventReaderForSegment(segment, config.getReadBufferSize());
-        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
-                                                                                                          segment.getStreamName()), RuntimeException::new);
-        DelegationTokenProvider delegationTokenProvider = DelegationTokenProviderFactory.create(delegationToken, controller, segment);
+        DelegationTokenProvider delegationTokenProvider = DelegationTokenProviderFactory.create(controller, segment,
```

Comment:
This is not read-only. The revisioned stream client is used for writes. 
This is called from the public API above, not for internal streams.

Please add unit / integration tests such that this sort of bug would be caught.

Reply:
Ok. I've modified the access operation to `read_write` to reflect that.

Reply:
Also, I've added a number of tests to supplement the many existing tests we already have for Auth.